### PR TITLE
Simple k8s example

### PR DESCRIPTION
**Pull request by dimm0**
Added a minimal single-pod example for deploying jitsi to kubernetes cluster

### CODE REVIEW

**Comment by sapkra** [comment]
@saghul recommended to have all examples in an `examples` directory.

```
examples
-- kubernetes
-- openshift
-- others
```

**Comment by saghul** [comment]
Thanks for this! I'll try to give it a test soon. @aaronkvanmeerten @jmacelroy can you PTAL too?

**Comment by HubbeKing** [comment]
Since the containers use a bunch of the same environment variables (`XMPP_DOMAIN`, `XMPP_AUTH_DOMAIN`, `XMPP_INTERNAL_MUC_DOMAIN`, `JVB_BREWERY_MUC`,`TZ`), it would be better to define them in a ConfigMap and pull them into the containers using envFrom and configMapRef. This way, you can have them all in one place, just like how the existing docker-compose.yml file pulls them all from a .env file.

Likewise, since you're already defining `JVB_AUTH_PASSWORD` and `JICOFO_AUTH_PASSWORD` in Secrets, you should also define `JVB_AUTH_USER` and `JICOFO_AUTH_USER` in those Secrets, to make it easier to adjust these values without making mistakes.

**Comment by jmacelroy** [comment]
> Since the containers use a bunch of the same environment variables (`XMPP_DOMAIN`, `XMPP_AUTH_DOMAIN`, `XMPP_INTERNAL_MUC_DOMAIN`, `JVB_BREWERY_MUC`,`TZ`), it would be better to define them in a ConfigMap and pull them into the containers using envFrom and configMapRef. This way, you can have them all in one place, just like how the existing docker-compose.yml file pulls them all from a .env file.
> 
> Likewise, since you're already defining `JVB_AUTH_PASSWORD` and `JICOFO_AUTH_PASSWORD` in Secrets, you should also define `JVB_AUTH_USER` and `JICOFO_AUTH_USER` in those Secrets, to make it easier to adjust these values without making mistakes.

I've typically kept secrets in secret store and not added associated non-secret config. Not every user can view the secrets and it seems like a trade-off. I'm fine leaving it for however the committer would like the example to be. In practice I imagine the example will always be slightly modified for the specific cluster.

Also, I think I could argue either way on the configmap issue in this case. As an example I think having the separate copy/paste makes it more clear what service uses what. In practice I'd probably templatize and use a single helm chart with the values.yaml defining the same for all env vars or separate configmaps per service.

Great thoughts and suggestions but I'm ok with implementer deciding in this case.

File: examples/kubernetes/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Install guide for kubernetes
+
+This guide will deploy jitsi in the most simple way: as several containers in a single pod. This is enough to start in case your hardware is enough. If you need to scale components to severa instance, you'll have to modify it to use several services and pods.
```
Comment (jmacelroy) [inline examples/kubernetes/README.md]: Small typo, severa->several instances.

**Comment by Bellusterra** [comment]
@dimm0 Thank you.

**Comment by dimm0** [comment]
Ok great!
Yeah, I tried to keep it as simple and minimal as possible, so that everyone can add to it

**Comment by Bellusterra** [comment]
Your instructions look very clear and easy to follow, which is great for a novice like myself :). Speaking of adding to it, I know the jibri docker was recently released and was thinking that would make a great addition... it might take me a month/year but would be happy to try and figure that out if it doesn’t exist already?

**Comment by dimm0** [comment]
It's actually on my todo list, and I started adding it to my install, but was stopped by a bunch of requirements like adding sound driver modules to the node.. Can it be simplified?
(we can talk in https://rocket.nautilus.optiputer.net if you want, to not pollute the thread here)

**Comment by dimm0** [comment]
I looked at jibri, and it doesn't seem very useful now... Only one session can be recorded, lots of pre-setup... I think I'll wait for something better to be implemented.

**Comment by aaronkvanmeerten** [comment]
Jibri was originally meant to be used on individual instances, so the docker image is only useful as a proof of concept I fear.  Part of the problem is the sound driver modules which need a device per jibri, as well as the CPU/memory required to run the chrome as a meeting participant.  I suggest running the jibris outside docker, and exposing the prosody client port to allow them to be discoverable, if you're looking to run more than a single recording.

**Comment by dimm0** [comment]
Why can't we simply dump the sound and video stream from jitsi core into a file? (I know nothing about how it's made inside unfortunately...)

**Comment by aaronkvanmeerten** [comment]
Absolutely a reasonable question.  The main answer is that the jitsi videobridge does no transcoding, and no formatting/rendering.  It is simply an SFU (selective forwarding unit).  So whatever is going to do the recording would need to receive all relevant packets, decrypt and render them. Also, since we get streams from each participant, we'd need to choose how to composite these into a screen.  The best tool for all of these jobs together is a browser (since we already do this in the jitsi-meet client), so that's how we got here.

**Comment by dimm0** [comment]
Seems like ffmpeg can do this...
https://unix.stackexchange.com/questions/233832/merge-two-video-clips-into-one-placing-them-next-to-each-other

Just need to run it with proper params depending on the number of participants, and pack in a container :)

**Comment by dimm0** [comment]
Moving discussion here https://github.com/jitsi/docker-jitsi-meet/issues/13

**Comment by AliMD** [comment]
@dimm0 I'm sure you dont test at all (Specially more than 2 users) 😞 
Its not work at all and I found many bugs in enviroment!

**Comment by AliMD** [comment]
@dimm0 I'm sure you dont test it at all (Specially more than 2 users) 😞
Its not work at all and I found many bugs in enviroment!

@jmacelroy Please dont merge before testing, I took more than 20 hours to fix it! And steel finding bugs in production!

**Comment by saghul** [comment]
@AliMD The examples (anything other than the docker-compose file) are 100% community maintained and are self-serving. If you found problems, please send a PR!

**Comment by dimm0** [comment]
Works fine for me with 10 people

**Pull request merged**